src/main/java/io/confluent/connect/elasticsearch/RetryUtil.java

```diff
@@ -86,5 +94,73 @@ public static long computeRetryWaitTimeInMillis(int retryAttempts,
     return result < 0L ? MAX_RETRY_TIME_MS : Math.min(MAX_RETRY_TIME_MS, result);
   }
 
+  /**
+   * Call the supplied function up to the {@param maxAttempts maximum number of attempts}.
+   *
+   * <p>The description of the function should be a succinct, human-readable present tense phrase
+   * that summarizes the function, such as "read tables" or "connect to database" or
+   * "make remote request". This description will be used within exception and log messages.
+   *
+   * @param description      present tense description of the action, used to create the error
+   *                         message; may not be null
+   * @param function         the function to call; may not be null
+   * @param maxAttempts      maximum number of attempts
+   * @param initialBackoff   the initial backoff in ms before retrying
+   * @return the function's return value
+   * @throws ConnectException        if the function failed after retries
+   */
+  public static <T> T callWithRetries(
+      String description,
+      Callable<T> function,
+      int maxAttempts,
+      long initialBackoff
+  ) throws Exception {
+    return callWithRetries(description, function, maxAttempts, initialBackoff, Time.SYSTEM);
+  }
 
+  /**
+   * Call the supplied function up to the {@param maxAttempts maximum number of attempts}.
+   *
+   * <p>The description of the function should be a succinct, human-readable present tense phrase
+   * that summarizes the function, such as "read tables" or "connect to database" or
+   * "make remote request". This description will be used within exception and log messages.
+   *
+   * @param description      present tense description of the action, used to create the error
+   *                         message; may not be null
+   * @param function         the function to call; may not be null
+   * @param maxAttempts      maximum number of attempts
+   * @param initialBackoff   the initial backoff in ms before retrying
+   * @param clock            the clock to use for waiting
+   * @return the function's return value
+   * @throws ConnectException        if the function failed after retries
+   */
+  protected static <T> T callWithRetries(
+      String description,
+      Callable<T> function,
+      int maxAttempts,
+      long initialBackoff,
+      Time clock
+  ) throws Exception {
+    int attempt = 0;
+    while (true) {
+      ++attempt;
+      try {
+        log.trace(
+            "Try {} (attempt {} of {})",
+            description,
+            attempt,
+            maxAttempts
+        );
+        return function.call();
+      } catch (IOException e) {
```

Comment:
If this is a generic util method, do we have to catch `Exception` instead?

Reply:
no because the client can throw runtime exceptions which should not be retried only http client exceptions should be retried